app/controllers/transactions/transaction-list.controller.js

```diff
@@ -16,50 +16,56 @@ const client = new ConnectorClient(process.env.CONNECTOR_URL)
 
 const { CORRELATION_HEADER } = require('../../utils/correlation-header.js')
 
-module.exports = (req, res) => {
+function error (req, res, msg) {
+  return renderErrorView(req, res, msg)
+}
+
+module.exports = async (req, res, next) => {
   const accountId = auth.getCurrentGatewayAccountId(req)
   const filters = getFilters(req)
 
   const correlationId = req.headers[CORRELATION_HEADER] || ''
   req.session.filters = url.parse(req.url).query
-  if (!filters.valid) return error('Invalid search')
 
-  transactionService
-    .search([ accountId ], filters.result, correlationId)
-    .then(transactions => {
-      client
-        .getAllCardTypes({ correlationId }, allCards => {
-          const model = buildPaymentList(transactions, allCards, accountId, filters.result, router.paths.transactions.download)
-          model.search_path = router.paths.transactions.index
-          model.filtersDescription = describeFilters(filters.result)
-          model.eventStates = states.allDisplayStateSelectorObjects()
-            .map(state => {
-              return {
-                value: state.key,
-                text: state.name,
-                selected: filters.result.selectedStates && filters.result.selectedStates.includes(state.name)
-              }
-            })
-          model.eventStates.unshift({ value: '', text: 'Any', selected: false })
+  if (!filters.valid) {
+    return error(req, res, 'Invalid search')
+  }
+
+  let result
```

Comment:
I'd be tempted to put the try/catch around the whole of the rest of the function so we don't need to declare using `let`. But there isn't a problem with it this way.